src/Php84/Resources/stubs/Pdo/Mysql.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Pdo;
+
+use PDO;
+use PDOException;
+
+if (\PHP_VERSION_ID < 80400) {
+    class Mysql extends PDO
+    {
+        public const ATTR_COMPRESS = PDO::MYSQL_ATTR_COMPRESS;
+        public const ATTR_DIRECT_QUERY = PDO::MYSQL_ATTR_DIRECT_QUERY;
+        public const ATTR_FOUND_ROWS = PDO::MYSQL_ATTR_FOUND_ROWS;
+        public const ATTR_IGNORE_SPACE = PDO::MYSQL_ATTR_IGNORE_SPACE;
+        public const ATTR_INIT_COMMAND = PDO::MYSQL_ATTR_INIT_COMMAND;
+        public const ATTR_LOCAL_INFILE = PDO::MYSQL_ATTR_LOCAL_INFILE;
+        public const ATTR_LOCAL_INFILE_DIRECTORY = \PHP_VERSION_ID >= 80100 ? \PDO::MYSQL_ATTR_LOCAL_INFILE_DIRECTORY : 1015;
+        public const ATTR_MAX_BUFFER_SIZE = PDO::MYSQL_ATTR_MAX_BUFFER_SIZE;
```

Comment:
https://www.php.net/manual/en/class.pdo-mysql.php#pdo-mysql.constants.attr-max-buffer-size

This constant (and also `ATTR_READ_DEFAULT_FILE` and `ATTR_READ_DEFAULT_GROUP`) are not available when `mysqlnd` is used. Unfortunately, this means an exception is thrown when a class instance is constructed in such cases.

Options:
- just use a numeric value rather than refer to potentially defined value
- find some way to fallback if not defined (probably impossible as what can be done within the constant declaration seems to be very limited)
- leave them out

Reply:
we can should define two classes, in an if/else then, to mimic what PHP does